engine/src/main/java/org/terasology/math/ChunkMath.java

```diff
@@ -30,79 +33,208 @@
     }
 
+    /**
+     *
+     * @param pos
+     * @return
+     * @deprecated This is scheduled for removal in an upcoming version
+     *             method will be replaced with JOML implementation {@link #calcChunkPos(Vector3ic, org.joml.Vector3i)}.
+     */
+    @Deprecated
     public static Vector3i calcChunkPos(Vector3i pos) {
         return calcChunkPos(pos.x, pos.y, pos.z);
     }
 
+    /**
+     *
+     * @param x
+     * @param y
+     * @param z
+     * @return
+     * @deprecated This is scheduled for removal in an upcoming version
+     *             method will be replaced with JOML implementation {@link #calcChunkPos(int, int, int, org.joml.Vector3i)}.
+     */
+    @Deprecated
     public static Vector3i calcChunkPos(int x, int y, int z) {
         return calcChunkPos(x, y, z, ChunkConstants.CHUNK_POWER);
     }
 
+    /**
+     *
+     * @param x
+     * @param y
+     * @param z
+     * @param chunkPower
+     * @return
+     * @deprecated This is scheduled for removal in an upcoming version
+     *             method will be replaced with JOML implementation {@link #calcChunkPos(int, int, int, org.joml.Vector3i)}.
+     */
+    @Deprecated
+    public static Vector3i calcChunkPos(int x, int y, int z, Vector3i chunkPower) {
+        return new Vector3i(calcChunkPos(x, chunkPower.x), calcChunkPos(y, chunkPower.y), calcChunkPos(z, chunkPower.z));
+    }
+
+    //TODO: can be replaced by region3i iterator
     public static Vector3i[] calcChunkPos(Region3i region) {
         return calcChunkPos(region, ChunkConstants.CHUNK_POWER);
     }
 
-    public static Vector3i calcChunkPos(int x, int y, int z, Vector3i chunkPower) {
-        return new Vector3i(calcChunkPosX(x, chunkPower.x), calcChunkPosY(y, chunkPower.y), calcChunkPosZ(z, chunkPower.z));
+    /**
+     * The position of the chunk given the coordinate and size of chunk in powers of 2.
+     *
+     * <p>default chunk size ({@link ChunkConstants#SIZE_X}, {@link ChunkConstants#SIZE_Y}, {@link ChunkConstants#SIZE_Z}) </p>
+     *
+     * @param pos absolute position of the block
+     * @param dest will hold the result
+     * @return dest
+     */
+    public static org.joml.Vector3i calcChunkPos(Vector3fc pos, org.joml.Vector3i dest) {
+        return calcChunkPos(Math.roundHalfUp(pos.x()), Math.roundHalfUp(pos.y()), Math.roundHalfUp(pos.z()), ChunkConstants.POWER_X, ChunkConstants.POWER_Y, ChunkConstants.POWER_Z, dest);
+    }
+
+    /**
+     * The position of the chunk given the coordinate and size of chunk in powers of 2.
+     *
+     * @param pos absolute position of the block
+     * @param chunkPower the size of the chunk in powers 2
+     * @param dest will hold the result
+     * @return dest
+     */
+    public static org.joml.Vector3i calcChunkPos(Vector3fc pos, Vector3ic chunkPower, org.joml.Vector3i dest) {
+        return calcChunkPos(Math.roundHalfUp(pos.x()), Math.roundHalfUp(pos.y()), Math.roundHalfUp(pos.z()),
```

Comment:
Are things pick about how these values are rounded?

I suggest making a test for it. As much for documentation as anything else, and it'll explicitly point out if this ever changes. It's one of those almost-too-simple-to-test things otherwise.

values like -1000f, -499.9f, -500.1f, -500.5f, -0f, 0f, 500.5f, 500.1f, -499.9f, 1000f
or somesuch for mix of positive, negative, above-half and below-half.

If they're _not_ picky about how they're rounded, x.intValue() might be ever-so-slightly cheaper. except that'll be always-towards-zero, which is probably worse for us than always-more-positive or always-more-negative, so, in conclusion, I'm over-thinking things again